src/main/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/controller/NomisSyncController.kt

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.digital.hmpps.visitallocationapi.controller
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import jakarta.validation.Valid
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.visitallocationapi.dto.nomis.VisitAllocationPrisonerMigrationDto
+import uk.gov.justice.digital.hmpps.visitallocationapi.dto.nomis.VisitAllocationPrisonerSyncDto
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+
+const val VO_PRISONER_MIGRATION: String = "/visits/allocation/prisoner/migrate"
+const val VO_PRISONER_SYNC: String = "/visits/allocation/prisoner/sync"
+
+@RestController
+class NomisSyncController {
+  @PreAuthorize("hasRole('VISIT_ALLOCATION_MIGRATION')")
+  @PostMapping(VO_PRISONER_MIGRATION)
+  @Operation(
+    summary = "Endpoint to migrate prisoner VO / PVO balances from NOMIS to DPS.",
+    description = "Takes a prisoner and 'onboards' them onto DPS, syncing their balance with NOMIS.",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Prisoner information has been migrated.",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint.",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to migrate prisoner VO / PVO information.",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun migratePrisonerVisitOrders(@RequestBody @Valid visitAllocationPrisonerMigrationDto: VisitAllocationPrisonerMigrationDto): ResponseEntity<Void> = ResponseEntity.status(HttpStatus.OK).build()
+
+  @PreAuthorize("hasRole('VISIT_ALLOCATION_SYNC')")
```

Comment:
do we want 2 different roles? and do we need to adhere to the new role name format?